app/workers/evss/disability_compensation_form/submit_form8940.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+module EVSS
+  module DisabilityCompensationForm
+    class SubmitForm8940
+      include Sidekiq::Worker
+      include JobStatus
+
+      FORM_ID = '21-8940' # form id for PTSD IU
+      DOC_TYPE = 'L149'
+
+      STATSD_KEY_PREFIX = 'worker.evss.submit_form8940'
+
+      # Sidekiq has built in exponential back-off functionality for retrys
+      # A max retry attempt of 10 will result in a run time of ~8 hours
+      # This job is invoked from 526 background job
+      RETRY = 10
+
+      sidekiq_options retry: RETRY
+
+      # This callback cannot be tested due to the limitations of `Sidekiq::Testing.fake!`
+      sidekiq_retries_exhausted do |msg, _ex|
+        Rails.logger.send(
+          :error,
+          "Failed all retries on SubmitForm8940 submit, last error: #{msg['error_message']}"
+        )
+        Metrics.new(STATSD_KEY_PREFIX, msg['jid']).increment_exhausted
+      end
+
+      # Performs an asynchronous job for generating and submitting 8940 PDF documents to VBMS
+      #
+      # @param auth_headers [Hash] The VAAFI headers for the user
+      # @param evss_claim_id [String] EVSS Claim id received from 526 submission
+      # @param saved_claim_id [Integer] Saved Claim id from 526 submission
+      # @param submission_id [String] The submission id of 526, uploads, and 4142 data
+      # @param form_content [Hash] The form content for 8940 submission
+      #
+      def perform(auth_headers, evss_claim_id, saved_claim_id, submission_id, form_content)
+        with_tracking('Form8940 Submission', saved_claim_id, submission_id) do
+          parsed_form = JSON.parse(form_content)
+          parsed_form8940 = parse_8940(parsed_form.deep_dup)
+
+          # process 8940
+          process_8940(auth_headers, evss_claim_id, parsed_form8940) if parsed_form8940.present?
```

Comment:
One of our goals in the job refactors was to keep jobs logic small (same theory as skinny controllers). Could the processing of the form (everything below `private`) be offloaded to some kind of document processor class that takes in a submission? The class could have accessors for `file_body`, and `data`. e.g.

```ruby
document = Form8940Document.new(submission)
document.data
document.file_body
```

`upload_to_vbms` would then be the only private method and take the output of the processor as it's input, e.g. 

```ruby
def upload_to_vbms(auth_headers, document)
  client = EVSS::DocumentsService.new(auth_headers)
  client.upload(document.file_body, document.data)
ensure
  # Delete the temporary PDF file
  File.delete(document.pdf_path) if document.pdf_path.present?
end
```

the job would then look like:

```ruby
def perform(submission_id)
  super(submission_id)
  with_tracking('Form8940 Submission', submission.saved_claim_id, submission.id) do
    upload_to_vbms(submission.auth_headers, Form8940Document.new(submission))
  end
rescue StandardError => error
  retryable_error_handler(error)
  raise error
end
```

This would separate the concern of building PDFs from sending them and I think have the side benefit of having more granular specs.

Reply:
That sounds good and makes total sense, but where should I add this?  It doesn't seem like something that would still live under `workers` but I'm not sure (and also new to Ruby!)

Reply:
@kreek - Does this match what what you were thinking?

Reply:
@dmccurdybah 👍yes, that looks great much cleaner :)

Reply:
Cool!  Let me know if you'd like any more changes or if this is good to go!